samples/bookinfo/gateway-api/bookinfo-gateway-demo.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: bookinfo-gateway
+  namespace: istio-system
+spec:
+  addresses:
+  - value: istio-ingressgateway.istio-system.svc.cluster.local
```

Comment:
I've added two bookinfo-gateway yaml files, this one is for the "Getting Started" doc which will create a Bookinfo Gateway bound to the istio-ingressgateway that is already installed as part of the `demo` profile in Getting Started. Other docs will use the second Bookinfo Gateway, below, that will use an automatically deployed default controller.

Reply:
^^^ @howardjohn Does that sound like a reasonable approach to you?

Reply:
No, the second one should just create an HTTPRoute. The model we should be pushing is that a Gateway config and deployment/IP/service are 1:1.

For advanced cases that isn't strictly true but that is the preference.

Reply:
@howardjohn, OK, that makes sense but I'm not sure you've answered my question. I'm wondering if setting `spec.addresses` to the istio-ingressgateway that is already deployed by the `demo` profile (in the Getting Started doc) is OK, or if we really shouldn't be setting `spec.addresses` in the docs because that should only ever be done in advanced use cases? I'm trying to keep the getting started doc as unchanged as possible so that users will feel that switching to gateway-api is not very different.

Reply:
We shouldn't be setting it in any demos since it should only be used for advanced cases. When using gateway API you probably want to us the minimal profile (disable gateway)